astropy/io/fits/hdu/compressed/header.py

```diff
@@ -659,25 +484,23 @@ def _image_header_to_bintable_header_and_coldefs(
     # number of fields in the table, the indicator for a compressed
     # image HDU, the data type of the image data and the number of
     # dimensions in the image data array.
-    bintable_header.set("NAXIS1", cols.dtype.itemsize, "width of table in bytes")
-    bintable_header.set(
+    bintable.header.set("NAXIS1", cols.dtype.itemsize, "width of table in bytes")
+    bintable.header.set(
         "TFIELDS", ncols, "number of fields in each row", after="GCOUNT"
     )
-    bintable_header.set(
+    bintable.header.set(
         "ZIMAGE", True, "extension contains compressed image", after=after
     )
-    bintable_header.set("ZBITPIX", zbitpix, bitpix_comment, after="ZIMAGE")
-    bintable_header.set(
-        "ZNAXIS", generated_image_header["NAXIS"], naxis_comment, after="ZBITPIX"
-    )
+    bintable.header.set("ZBITPIX", zbitpix, bitpix_comment, after="ZIMAGE")
+    bintable.header.set("ZNAXIS", image_header["NAXIS"], naxis_comment, after="ZBITPIX")
 
     # Strip the table header of all the ZNAZISn and ZTILEn keywords
```

Comment:
Since you create a new bintable, I think this can all go.

Reply:
Done